subprojects/language-java/src/main/java/org/gradle/language/java/plugins/JavaLanguagePlugin.java

```diff
@@ -1,228 +0,0 @@
-
-import com.google.common.collect.ImmutableSet;
-import org.gradle.api.DefaultTask;
-import org.gradle.api.Incubating;
-import org.gradle.api.Plugin;
-import org.gradle.api.Project;
-import org.gradle.api.Task;
-import org.gradle.api.artifacts.dsl.RepositoryHandler;
-import org.gradle.api.internal.artifacts.ArtifactDependencyResolver;
-import org.gradle.api.internal.artifacts.ImmutableModuleIdentifierFactory;
-import org.gradle.api.internal.artifacts.repositories.ResolutionAwareRepository;
-import org.gradle.api.internal.attributes.AttributesSchemaInternal;
-import org.gradle.internal.Transformers;
-import org.gradle.internal.deprecation.DeprecationLogger;
-import org.gradle.internal.operations.BuildOperationExecutor;
-import org.gradle.internal.service.ServiceRegistry;
-import org.gradle.jvm.JvmByteCode;
-import org.gradle.jvm.internal.JarBinarySpecInternal;
-import org.gradle.jvm.internal.JvmAssembly;
-import org.gradle.jvm.internal.WithDependencies;
-import org.gradle.jvm.internal.WithJvmAssembly;
-import org.gradle.jvm.internal.resolve.DefaultVariantsMetaData;
-import org.gradle.jvm.internal.resolve.SourceSetDependencyResolvingClasspath;
-import org.gradle.jvm.internal.resolve.VariantsMetaData;
-import org.gradle.jvm.platform.JavaPlatform;
-import org.gradle.language.base.DependentSourceSet;
-import org.gradle.language.base.LanguageSourceSet;
-import org.gradle.language.base.internal.SourceTransformTaskConfig;
-import org.gradle.language.base.internal.registry.LanguageTransform;
-import org.gradle.language.base.internal.registry.LanguageTransformContainer;
-import org.gradle.language.base.plugins.ComponentModelBasePlugin;
-import org.gradle.language.java.JavaSourceSet;
-import org.gradle.language.java.internal.DefaultJavaLanguageSourceSet;
-import org.gradle.language.java.tasks.PlatformJavaCompile;
-import org.gradle.language.jvm.plugins.JvmResourcesPlugin;
-import org.gradle.model.Mutate;
-import org.gradle.model.RuleSource;
-import org.gradle.model.internal.manage.schema.ModelSchema;
-import org.gradle.model.internal.manage.schema.ModelSchemaStore;
-import org.gradle.platform.base.BinarySpec;
-import org.gradle.platform.base.ComponentType;
-import org.gradle.platform.base.DependencySpec;
-import org.gradle.platform.base.TypeBuilder;
-import org.gradle.platform.base.internal.BinarySpecInternal;
-
-import java.io.File;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-
-import static com.google.common.collect.Iterables.concat;
-import static org.gradle.util.CollectionUtils.collect;
-import static org.gradle.util.CollectionUtils.first;
-
-/**
- * Plugin for compiling Java code. Applies the {@link org.gradle.language.base.plugins.ComponentModelBasePlugin} and {@link org.gradle.language.jvm.plugins.JvmResourcesPlugin}. Registers "java"
- * language support with the {@link JavaSourceSet}.
- *
- * @since 3.4
- */
-@Incubating
-@Deprecated
-public class JavaLanguagePlugin implements Plugin<Project> {
```

Comment:
Do we know if any plugins (Nebula? LinkedIn?) have references to this type like some plugins had with MavenPlugin?

Reply:
So far I haven't seen any references outside of "playground" projects that have been abandoned for several years: https://github.com/search?l=Java&p=1&q=org.gradle.language.java.plugins.JavaLanguagePlugin&type=Code

Reply:
We have already brought this up when we deprecated this with Netflix. They had never heard of these plugins. I think it's fair to assume that no one every used these for real. Because we cancelled "software model" on JVM before it took off in any way.